### PR TITLE
Disable the request token check for idempotent actions

### PR DESCRIPTION
**Pull request by aschempp**
FINALLY removes the request token check for idempotent actions in the back end.

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]
❤️

**Comment by leofeyer** [comment]
This reenables the CSRF vulnerability that we had closed in https://github.com/contao/core/issues/4007, doesn‘t it? @ausi /cc

**Comment by fritzmg** [comment]
The original issue was about editing and deleting. The request token check is still in place for `delete` and `toggle` actions and for POST requests.

**Comment by Toflar** [comment]
Anything we know is idempotent, does not need a token.

**Comment by ausi** [comment]
Are we sure, that `act=edit`, `act=show` and `act=select` are always idempotent?

**Comment by Toflar** [comment]
Yes, that's the whole purpose of this PR :D

**Comment by aschempp** [comment]
> Are we sure, that `act=edit`, `act=show` and `act=select` are always idempotent?

Technically, someone _could_ manipulate data in the `onload` callbacks. But then again you could also manipulate data in the list/parent/tree view where there is no request token check.

File: core-bundle/contao/drivers/DC_Table.php [hunk 1]
```diff
@@ -116,13 +116,13 @@ public function __construct($strTable, $arrModule=array())
...
-		if (Input::get('act') !== null)
-		{
-			if (Input::get('rt') === null || !$container->get('contao.csrf.token_manager')->isTokenValid(new CsrfToken($container->getParameter('contao.csrf_token_name'), Input::get('rt'))))
-			{
-				$objSession->set('INVALID_TOKEN_URL', Environment::get('requestUri'));
-				$this->redirect($container->get('router')->generate('contao_backend_confirm'));
-			}
+		if (
+			Input::get('act') !== null
+			&& !\in_array(Input::get('act'), array('edit', 'show', 'select'), true)
```
Comment (ausi) [inline core-bundle/contao/drivers/DC_Table.php]:
```suggestion
			!\in_array(Input::get('act'), array(null, 'edit', 'show', 'select'), true)
```

**Review by ausi — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Pull request merged**